content/java/concepts/math-methods/terms/ulp/ulp.md

```diff
@@ -0,0 +1,79 @@
+---
+Title: '.ulp()'
+Description: 'Returns the size of the ulp of a number which is its unit of last precision.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Methods'
+  - 'Arithmetic'
+  - 'Documentation'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **Math.ulp()** method returns the unit of least precision of a given number. It calculates the distance between the given `float` or `double` value and the `float` `double` value next larger in magnitude.
+
+## Syntax
+
+```pseudo
+Math.ulp(num);
+```
+
+- The `num` parameter can either be of type `float` or `double`.
+- The return type will be the same as the parameter type.
+
+Special cases include the following:
+
+- If `num` is NaN `ulp(num)` will be NaN.
+- If `num` is positive or negative Zero then `ulp(num)` will be `Double.MAX_VALUE` or `Float.MAX_VALUE` meaning the minimum numbers that `float` or `double` can represent.
+- If `num` is positive or negative infinity `ulp(num)` will be postive infinity.
+- If `num` is a number `ulp(num)` is equal to `ulp(-num)`.
+- If the argument is `Double.MAX_VALUE` or `Float.MAX_VALUE` `ulp(num)` will be 2<sup>971</sup> for double and 2<sup>104</sup> for float.
```

Comment:
Rather than stating the method itself each time, let's refer to it as the return value

Reply:
I agree it's more clear and readable